core/carp_string.h

```diff
@@ -195,3 +265,24 @@ String Long_format(String* str, long x) {
 long Long_from_MINUS_string(String *s) {
     return atol(*s);
 }
+
+int String_index_MINUS_of_MINUS_from(String *s, char c, int i) {
```

Comment:
Couldn’t this be expressed in Carp using `String.chars` and then iterating over the array? It would have worse runtime characteristics, though...

Reply:
I initially wrote it in Carp, but AFAIK Carp lacks an explicit 'return' statement.
This meant exiting from a for/while loop early in Carp (early return) wasn't as neat.

Reply:
There is `break` if that helps. We have some functions in Core that use that in similar cases.

Reply:
ahh okay, I was looking for 'return'
I want to avoid String.chars as IIRC it is a copy, but I can always use String.chat-at.

Can we land this C version and I will try follow up later with a Carp version ?

Reply:
Sounds good!